tests/acceptance/bundle/BundleCustomFieldsCest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * Tests that subscribers are added to Kit and custom field
+ * data assigned based on the assigned bundle.
+ *
+ * @since   1.2.8
+ */
+class BundleCustomFieldsCest
+{
+	/**
+	 * Run common actions before running the test functions in this class.
+	 *
+	 * @since   1.2.8
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function _before(AcceptanceTester $I)
+	{
+		// Activate Plugins.
+		$I->activateConvertKitPlugin($I);
+		$I->activateThirdPartyPlugin($I, 'membermouse-platform');
+		$I->memberMouseSetupPlugin($I);
+	}
+
+	/**
+	 * Test that the custom field data is stored against the subscriber
+	 * in Kit when a bundle is assigned to the member in MemberMouse.
+	 *
+	 * @since   1.2.8
+	 *
+	 * @param   AcceptanceTester $I  Tester.
+	 */
+	public function testMemberCustomFieldsWhenBundleAdded(AcceptanceTester $I)
+	{
+		// Create a product.
+		$productID = $I->memberMouseCreateProduct($I, 'Product', $_ENV['MEMBERMOUSE_PRODUCT_REFERENCE_KEY']);
+
+		// Create bundle.
+		$bundleID = $I->memberMouseCreateBundle($I, 'Bundle', [ $productID ]);
+
+		// Setup Plugin to tag users purchasing the bundle to the
+		// Kit Tag ID, and store the Last Name in the Kit
+		// Last Name Custom Field.
+		$I->setupConvertKitPlugin(
+			$I,
+			[
+				'convertkit-mapping-bundle-' . $bundleID => $_ENV['CONVERTKIT_API_TAG_ID'],
+				'custom_field_last_name'                 => 'last_name',
+			]
+		);
+
+		// Generate email address for test.
+		$emailAddress = $I->generateEmailAddress();
+
+		// Logout.
+		$I->memberMouseLogOut($I);
+
+		// Complete checkout.
+		$I->memberMouseCheckoutProduct($I, $_ENV['MEMBERMOUSE_PRODUCT_REFERENCE_KEY'], $emailAddress);
+
+		// Check subscriber exists.
+		$subscriber = $I->apiCheckSubscriberExists($I, $emailAddress);
+
+		// Check that the subscriber has been assigned to the tag.
+		$I->apiCheckSubscriberHasTag($I, $subscriber['id'], $_ENV['CONVERTKIT_API_TAG_ID']);
+
+		// Check that the subscriber has the custom field data.
+		$I->apiCustomFieldDataIsValid(
+			$I,
+			$subscriber,
+			[
+				'last_name' => 'Last',
```

Comment:
Where is this value coming from?

Reply:
It's set when creating a new member in MemberMouse:
https://github.com/Kit/convertkit-membermouse/blob/last-name-custom-field/tests/_support/Helper/Acceptance/MemberMouse.php#L114